### ISSUE TITLE
errors: Add exceptions for commonly occurring response errors

### ISSUE THREAD

**Issue by harshavardhana** [minio/minio-py#492]
For example right now we have ResponseError i.e which leads to code like 

```
try:
   fn()
except ResponseError as err;
   if err.code == 'NoSuchBucket':
```

This coding can lead to bugs if not written properly. 

Add standard errors for known error responses from S3 for example 

- NoSuchBucket
- NoSuchKey
... 

http://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html

**Comment by harshavardhana** [comment]
Thanks @dcharbonnier for pointing this out.

**Comment by harshavardhana** [comment]
Working with @MrRedAmber https://github.com/minio/minio-py/pull/498 on addressing this issue.  ETA should be by Tuesday.

**Comment by harshavardhana** [comment]
Fixed by a07ed8ac1b6e020d2c9f31fa46c7e18be76970a5
